File: Projekt/Seans.cpp
#include "Seans.h"
#include <iostream>
int Seans::number_of_seans=0;
Seans::Seans():Film()
{
	number_of_seans++;
	hour.g=0;
	hour.m=0;
	day.d=0;
	day.m=0;
	day.y=0;
	type_seans="";
	sala=NULL;
}
Seans::Seans(string title,int time,int restrictions,int id_film, ::hour hour,::day day, Sala sala,string type_seans):Film(title,time,restrictions,id_film),hour(hour),day(day), type_seans(type_seans)
{
	number_of_seans++;
	this->sala=new Sala(sala);
}
Seans::Seans(const Film &film,::hour hour,::day day,Sala sala,string type_seans):Film(film),hour(hour),day(day),type_seans(type_seans)
{
	number_of_seans++;
	this->sala=new Sala(sala);
}
Seans::~Seans()
{
	number_of_seans--;
	if (sala!=NULL)
		delete sala;
}
void Seans::set_hour(::hour hour)
{
	this->hour=hour;
}
void Seans::set_sala(Sala sala)
{
	if(!(this->sala==NULL))
		delete this->sala;
	this->sala=new Sala(sala);
}
void Seans::set_type_seans(string type_seans)
{
	this->type_seans=type_seans;
}
void Seans::set_day(::day day)
{
	this->day=day;
}
::day Seans::get_day()
{
	return day;
}
::hour Seans::get_hour()
{
	return hour;
}
int Seans::get_number_of_hall()
{
	return sala->get_number();
}
Sala Seans::get_sala()
{
	return *sala;
}
Sala& Seans::get_sala2()
{
	return *this->sala;
}
string Seans::get_type_seans()
{
	return type_seans;
}
ostream& operator<< (ostream &output, Seans const& seans)
{
	output<<seans.day.d<<"."<<seans.day.m<<"."<<seans.day.y<<" "<<seans.hour.g<<":"<<seans.hour.m<<"  "<<static_cast<const Film&>(seans)<<"   "<<seans.type_seans<<"  sala: "<<(seans.sala->get_number())+1<<endl;
	return output;
}
int Seans::get_number_of_seans()
{
	return number_of_seans;
}
void Seans::save(ofstream &ofs)
{
	Film::save(ofs);
	int x;
	x=type_seans.length();
	ofs.write((char*)(&x), sizeof(int));
	ofs.write((char*)(type_seans.c_str()), x);
	ofs.write((char*)(&hour.g), sizeof(int));
	ofs.write((char*)(&hour.m), sizeof(int));
	ofs.write((char*)(&day.d), sizeof(int));
	ofs.write((char*)(&day.m), sizeof(int));
	ofs.write((char*)(&day.y), sizeof(int));
	sala->save(ofs);
}
void Seans::read(ifstream &ifs)
{
	Film::read(ifs);
	int x;
	char* temp = new char[sizeof(int)];
	ifs.read(temp, sizeof(int));
	x=*(int*)(temp);
	delete temp;
	temp = new char[x];
	ifs.read(temp, x);
	for(int i=0;i<x;i++)
		type_seans=type_seans+(temp[i]);
	delete temp;
	temp=new char[sizeof(int)];
	ifs.read(temp, sizeof(int));
	hour.g=*(int*)(temp);
	ifs.read(temp, sizeof(int));
	hour.m=*(int*)(temp);
	ifs.read(temp, sizeof(int));
	day.d=*(int*)(temp);
	ifs.read(temp, sizeof(int));
	day.m=*(int*)(temp);
	ifs.read(temp, sizeof(int));
	day.y=*(int*)(temp);
	delete temp;
	if (sala==NULL)
		sala= new Sala();
	sala->read(ifs);
}

File: Projekt-java/Seans.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Maciej
 */
public class Seans extends Film {
    private Day day;
    private Hour hour;
    private Sala sala;
    private String type_seans;
    private static int number_of_seans=0;
    
    public Seans() {
        //day= new day()
        day=new Day(0,0,0);
        hour = new Hour(0,0);
        sala=null;
        type_seans="";
        number_of_seans++;
    }
    public Seans(Film film,int g,int m,int dd,int md,int dy,Sala sala, String type_seans){
        super(film);
        this.day=new Day(dd,md,dy);
        this.hour= new Hour(g,m);
        this.sala=new Sala(sala);
        this.type_seans=type_seans;
        number_of_seans++; 
    }
    public Seans(Seans seans){
        this.day=seans.day;
        this.hour=seans.hour;
        this.sala=seans.sala;
        this.type_seans=seans.type_seans;
        number_of_seans++;
    }
    @Override
    protected void finalize(){
        sala=null;
        hour=null;
        day=null;
        number_of_seans--;
    }
    
    public static int get_number_of_seans(){
        return number_of_seans;
    }
	
    public void set_hour(Hour hour){
        this.hour.set_hour(hour.get_hour_g(),hour.get_hour_m());

    }
    public void set_sala(Sala sala){
        this.sala=null;
        this.sala=new Sala(sala);
    }
    public void set_type_seans(String type_seans){
        this.type_seans=type_seans;
    }
    public void set_day(Day day){
        this.day.set_day(day.get_day_d(), day.get_day_m(), day.get_day_y());
    }
    public Day get_day(){
        return day;
    }
    public Hour get_hour(){
        return hour;
    }
    public int get_number_of_hall(){
        return sala.get_number();
    }
    public Sala get_sala(){
        return sala;
    }
    public String get_type_seans(){
        return type_seans;
    } 
    public void view_seans(){
        System.out.print(day.get_day_d()+"."+day.get_day_m()+"."+day.get_day_y()+"r ");
        System.out.print(hour.get_hour_g()+":"+hour.get_hour_m()+" ");
        System.out.print(this.get_title()+" "+this.type_seans+" "+this.get_time()+"min +"+this.get_restrictions()+" ");
        System.out.print("sala:"+(sala.get_number()+1));
    }
}
